feat: scroll to show the new active tab header

anwarulislam opened the pull request:
### Description

Improved the tab head section. When a tab selection is updated automatically (like when we click on a request in a collection which has a tab open or we click on add new tab button), the tab headers section should scroll to show the new active tab header.


### Walkthrough
* Import and use `nextTick` function from `vue` to manipulate scroll position of tab container based on active tab ID ([link](https://github.com/hoppscotch/hoppscotch/pull/3013/files?diff=unified&w=0#diff-052acc17f6ece720f660d3841f7e49f0a3a1d6ad68dc973819e3d758e45141d5L134-R142), [link](https://github.com/hoppscotch/hoppscotch/pull/3013/files?diff=unified&w=0#diff-052acc17f6ece720f660d3841f7e49f0a3a1d6ad68dc973819e3d758e45141d5R307-R333))
* Define and use `TAB_WIDTH` constant to calculate tab container width and tab position ([link](https://github.com/hoppscotch/hoppscotch/pull/3013/files?diff=unified&w=0#diff-052acc17f6ece720f660d3841f7e49f0a3a1d6ad68dc973819e3d758e45141d5L189-R200))
* Add `watch` effect to monitor changes of `modelValue` prop and scroll to active tab if needed ([link](https://github.com/hoppscotch/hoppscotch/pull/3013/files?diff=unified&w=0#diff-052acc17f6ece720f660d3841f7e49f0a3a1d6ad68dc973819e3d758e45141d5R307-R333))

AndrewBastin reviewed (changes requested):
The scroller doesn't reflect the jump correctly

Here, I try to open a tab which is placed at the end of the tabs list.


https://user-images.githubusercontent.com/9131943/233834281-5cbe0fb1-a620-4305-8771-e6393ab401b0.mov

nivedin reviewed (changes requested):
https://user-images.githubusercontent.com/53208152/235628053-f0b88fce-2dc6-4cec-b13f-c0cc2a4633fb.mp4

the scroll-to-active-tab does not work all the time...it gets glitchy sometimes and I am unable to scroll after that.

anwarulislam commented:
@amk-dev, @nivedin 

I have updated the solution with native scrolling API. And implemented `intersectionObserver` to observe the scrolling ended to update `thumbPosition`

You guys can take a look.

liyasthomas reviewed (approved):
LGTM 💯

nivedin reviewed (approved):
LGTM 💯💯
